public/javascript/index.js

```diff
@@ -40,4 +66,33 @@ function toggleChat(evt, chatName) {
     }
 }
 
-window.onload = initialize();
\ No newline at end of file
+
+// Given a user identifier, creates a new chat for that user if the identifier is unique
+// and logs an error if it is a duplicate
+function newChat(user) {
+    validUser = true;
+    for (chat of chats) {
+        if (user == chat.user) {
+            console.log(Error('Cannot have multiple chats with identical user identifiers'));
+            validUser = false;
+        }
+    }
+    if (validUser) {
+        chats.push({ user: user, messageStream: { messages: [], active: true } });
+    }
+}
+
```

Comment:
Since this requires a messageObject as a parameter, can you add a function called newMessageObject that takes all necessary components and creates that object?